pkg/operator/controller/controller_dns_daemonset.go

```diff
@@ -230,6 +230,58 @@ func daemonsetConfigChanged(current, expected *appsv1.DaemonSet) (bool, *appsv1.
 	return true, updated
 }
 
+// cmpConfigMapVolumeSource compares two configmap volume source values and
+// returns a Boolean indicating whether they are equal.
+func cmpConfigMapVolumeSource(a, b corev1.ConfigMapVolumeSource) bool {
+	if a.LocalObjectReference != b.LocalObjectReference {
+		return false
+	}
+	if !cmp.Equal(a.Items, b.Items, cmpopts.EquateEmpty()) {
+		return false
+	}
+	aDefaultMode := int32(420)
+	if a.DefaultMode != nil {
+		aDefaultMode = *a.DefaultMode
+	}
+	bDefaultMode := int32(420)
+	if b.DefaultMode != nil {
+		bDefaultMode = *b.DefaultMode
+	}
+	if aDefaultMode != bDefaultMode {
+		return false
+	}
+	if !cmp.Equal(a.Optional, b.Optional, cmpopts.EquateEmpty()) {
+		return false
+	}
+	return true
+}
+
+// cmpSecretVolumeSource compares two secret volume source values and returns a
+// Boolean indicating whether they are equal.
+func cmpSecretVolumeSource(a, b corev1.SecretVolumeSource) bool {
+	if a.SecretName != b.SecretName {
+		return false
+	}
+	if !cmp.Equal(a.Items, b.Items, cmpopts.EquateEmpty()) {
+		return false
+	}
+	aDefaultMode := int32(420)
```

Comment:
Would it make sense to define a `defaultMode := int32(420)` literal somewhere, in the event the default mode were to  change?

Reply:
I think it is extremely unlikely that the default mode would change because that seems like it would be a breaking API change.  I could introduce a constant though.  Would using the same constant for both secret volume sources and configmap volume sources make sense?

Reply:
Ah, my mistake. I think using a constant as you mentioned would make sense.

Reply:
Updated.  Thanks!